crates/sui-framework/packages/move-stdlib/sources/type_name.move

```diff
@@ -35,13 +51,38 @@ module std::type_name {
     /// later upgrade).
     public native fun get_with_original_ids<T>(): TypeName;
 
+    /// Returns true iff the TypeName represents a primitive type, i.e. one of
+    /// u8, u16, u32, u64, u128, u256, bool, address, vector.
+    public fun is_primitive(self: &TypeName): bool {
+        let bytes = ascii::as_bytes(&self.name);
+        bytes == &b"bool" ||
+        bytes == &b"u8" ||
+        bytes == &b"u16" ||
+        bytes == &b"u32" ||
+        bytes == &b"u64" ||
+        bytes == &b"u128" ||
+        bytes == &b"u256" ||
+        bytes == &b"address" ||
+        (vector::length(bytes) >= 6 &&
+        *vector::borrow(bytes, 0) == ASCII_V &&
+        *vector::borrow(bytes, 1) == ASCII_E &&
+        *vector::borrow(bytes, 2) == ASCII_C &&
+        *vector::borrow(bytes, 3) == ASCII_T &&
+        *vector::borrow(bytes, 4) == ASCII_O &&
+        *vector::borrow(bytes, 5) == ASCII_R)
```

Comment:
This makes me sad :'( But not much to do other than this

Reply:
It makes me sad, but I like how it lines up at least :P 

V
E
C
T
O
R